unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/Interfaces/EntityIdHelper.cs

```diff
@@ -35,12 +39,57 @@ public long EntityFromLegacyEntityString(string entityId)
                     return (long) SpecialEntityId.THIRD_PERSON_CAMERA_ENTITY_REFERENCE;
             }
 
-            long entityIdLong = entityId.GetHashCode() << 9;
-
+            long entityIdLong = DecodeBase36(entityId) << 9;
+            
             if (!entityIdToLegacyId.ContainsKey(entityIdLong))
                 entityIdToLegacyId[entityIdLong] = entityId;
 
             return entityIdLong;
         }
+        
+        public Int64 DecodeBase36(string input)
+        {
+            string charList = "0123456789abcdefghijklmnopqrstuvwxyz";
+            var reversed = input.ToLower().Reverse();
```

Comment:
We can boost performance by avoiding the `Reverse` using a loop from end to start.

Reply:
x2

Reply:
+10